crates/engine/tree/src/tree/mod.rs

```diff
@@ -2336,18 +2336,45 @@ where
         &mut self,
         block: RecoveredBlock<N::Block>,
     ) -> Result<InsertPayloadOk, InsertBlockError<N::Block>> {
-        self.insert_block_inner(block.clone())
-            .map_err(|kind| InsertBlockError::new(block.into_sealed_block(), kind))
+        match self.insert_block_inner(block) {
+            Ok(result) => Ok(result),
+            Err((kind, block)) => Err(InsertBlockError::new(block.into_sealed_block(), kind)),
+        }
```

Comment:
I'm wondering if the better approach for this change would be to remove the block context from `InsertBlockError` instead?

wdyt @Rjected